### PR TITLE
add an auto-hpo tool for data-recipes based on 3-sigma rule

### PR DESCRIPTION
**Pull request by yxdyc**
- implemented an auto-hpo tool for data-recipes based on 3-sigma rule, #136 ,
a log example is as follow:
image

- added a helper class to build the mapping from OPs' op_name to their used stats_key
- added a config exporter

### CODE REVIEW

**Comment by yxdyc** [comment]
some information about unit-test: 

passed in local machine (dlib==19.24.2), but failed in github
image

image

File: data_juicer/config/config.py [hunk 1]
```diff
@@ -478,6 +490,37 @@ def display_config(cfg):
...
+        save the config object, some params are from jsonargparse
+    :param cfg: cfg object to save (Namespace type)
+    :param path: the save path
+    :param format: 'yaml', 'json', 'json_indented', 'parser_mode'
+    :param skip_none: Whether to exclude entries whose value is None.
+    :param skip_check: Whether to skip parser checking.
+    :param overwrite: Whether to overwrite existing files.
+    :param multifile: Whether to save multiple config files
+        by using the __path__ metas.
+    :param branch:
```
Comment (HYLcool) [inline data_juicer/config/config.py:505]: Need to add description to this arg. BTW, it seems that save() function doesn't have an argument of "branch", maybe recheck is needed.

File: tools/hpo/README.md [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+$$P(|x-\mu| > 3\sigma) \leq 0.003$$
+
+To automate this process, we provide the tool which can be used as follows:
+```shell
+# cd tools/hpo
+python execute_hpo_3sigma.py --config <data-process-cfg-file-path>
+
+#e.g.,
+python execute_hpo_3sigma.py --config configs/process.yaml
+```
```
Comment (HYLcool) [inline tools/hpo/README.md:18]: The tool usage code might need to be aligned with the ZH version.

File: tools/hpo/execute_hpo_3sigma.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                        new_val = stats_key_to_mean[stats_key] - \
+                                  3 * stats_key_to_std[stats_key]
+                    if 'max' in arg_name:
+                        new_val = stats_key_to_mean[stats_key] + \
+                                  3 * stats_key_to_std[stats_key]
+                    if new_val is not None:
+                        logger.info(f'Using 3-sigma rule, changed para '
+                                    f'{arg_name}={args[arg_name]} into '
+                                    f'{arg_name}={new_val}')
+                        args[arg_name] = new_val
```
Comment (HYLcool) [inline tools/hpo/execute_hpo_3sigma.py:56]: There might be "nan" in mean and std for some string stats or other object stats (e.g. lang). Maybe a type check should be done before calculate the new_val.

**Review by HYLcool — APPROVED** [review]
LGTM

**Pull request merged**